src/Unosquare.RaspberryIO.Peripherals/Accelerometer GY521/AccelerometerGY521.cs

```diff
@@ -0,0 +1,154 @@
+        private const int AcclOutputY = 0x3d;
+        private const int AcclOutputZ = 0x3f;
+        private const int TemperatureOutput = 0x41;
+        private const int GyroConfig = 0x1b;
+        private const int AcclConfig = 0x1c;
+        private readonly Thread ReadWorker;
+        private readonly TimeSpan ReadTime = TimeSpan.FromSeconds(0.5);
+
+        /// <summary>
+        /// Initializes a new instance of the <see cref="AccelerometerGY521"/> class.
+        /// </summary>
+        /// <param name="device"> i2C device. </param>
+        /// <param name="gyroSens">The gyroscope sensitivity factor.</param>
+        /// <param name="acclSens">The accelerometer sensitivity factor.</param>
+        public AccelerometerGY521(II2CDevice device, FSSEL gyroSens, AFSSEL acclSens)
+        {
+            Device = device;
+            var gyroScale = (int)gyroSens << 3;
+            var acclScale = (int)acclSens << 3;
+            Device.WriteAddressByte(GyroConfig, (byte)gyroScale);
+            Device.WriteAddressByte(AcclConfig, (byte)acclScale);
+            ReadWorker = new Thread(Run);
+        }
+
+        /// <summary>
+        /// Occurs when [data available].
+        /// </summary>
+        public event EventHandler<AccelerometerGY521EventArgs> DataAvailable;
+
+        /// <summary>
+        /// Retrieves the I2C Device.
+        /// </summary>
+        public II2CDevice Device { get; }
+
+        /// <summary>
+        /// Gets a value indicating whether this instance is running.
+        /// </summary>
+        /// <value><c>true</c> if this instance is running; otherwise, <c>false</c>.</value>
+        public bool IsRunning { get; private set; }
+
+        /// <summary>
+        /// Starts the instance.
+        /// </summary>
+        public void Start()
+        {
+            // Reset sensor sleep mode to 0.
+            Device.WriteAddressByte(PwrMgmt1, 0);
+            IsRunning = true;
+            ReadWorker.Start();
+        }
+
+        /// <summary>
+        /// Stops this instance.
+        /// </summary>
+        public void Dispose()
+        {
+            if (IsRunning == false)
+                return;
+            StopContinuousReads();
+            SetSleepMode();
+        }
+
+        /// <summary>
+        /// Retrieve the data capted by the sensor.
+        /// </summary>
+        /// <returns> Data calculated by GY-521 accelerometer. </returns>
+        public AccelerometerGY521EventArgs RetrieveSensorData()
+        {
+            // Read registry for gyroscope and accelerometer on the three axis.
+            var gyro = new Point3d(ReadWord2C(GyroOutputX), ReadWord2C(GyroOutputY), ReadWord2C(GyroOutputZ));
+            var accel = new Point3d(ReadWord2C(AcclOutputX), ReadWord2C(AcclOutputY), ReadWord2C(AcclOutputZ));
+            var temperature = (ReadWord2C(TemperatureOutput) / 340.0) + 36.53;
+            var gyroSens = gyro / Device.ReadAddressByte(GyroConfig);
```

Comment:
You are performing the division against the raw value read from the registry, that's wrong because you need to read the value from the registry, extract the value of the FS_SEL bits and map that value to the actual scale factor. It seems like you just delete the array with the scale values.